yamcs-core/src/main/java/org/yamcs/tctm/UdpTcDataLink.java

```diff
@@ -37,9 +37,31 @@ protected void startUp() throws SocketException, UnknownHostException {
         socket = new DatagramSocket();
     }
 
+    @Override
+    public void doDisable() {
+        if (socket != null) {
+            socket.close();
+            socket = null;
+        }
+        if (thread != null) {
+            thread.interrupt();
+        }
+    }
+
+    @Override
```

Comment:
I think we don't need this override. 
Please move the thread.setName in the AbstractThreadedTcDataLink.doEnable